tests/components/iron_os/test_config_flow.py

```diff
@@ -32,10 +37,31 @@ async def test_form(
     assert len(mock_setup_entry.mock_calls) == 1
 
 
+@pytest.mark.usefixtures("discovery")
+async def test_async_step_user_devices_already_setup(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test we can't start a flow if there is already a config entry."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] is FlowResultType.FORM
+
+    config_entry.add_to_hass(hass)
```

Comment:
Let's first add it and then execute the flow, just a style thing

Reply:
That doesn't work, the user flow will instantly drop all discoveries from devices already configured and abort with no_devices_found if there aren't any other discoveries, so the only way to start a user flow that will abort because the entry is already configured, would be to first start the flow, then add the config entry and then submit the form for the already configured device.

That's the same way other BLE integrations test this case.